include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -36,24 +38,98 @@ class sycl_backend
     using resource_container_t = std::vector<execution_resource_t>;
 
   private:
-    class async_waiter
+    static inline bool is_profiling_enabled = false;
+    using report_clock_type = std::chrono::steady_clock;
+    using report_duration = std::chrono::milliseconds;
+
+    class async_waiter_base
+    {
+      public:
+        virtual void
+        report() = 0;
+        virtual bool
+        is_complete() = 0;
```

Comment:
I think we are able and should declare this function as `const` :
```suggestion
        is_complete() const = 0;
```
Here and in all required places.

Reply:
Fixed for `is_complete` and` report`